From e9078ad9508ef765d9855103f9833558f679a09d Mon Sep 17 00:00:00 2001
From: Liu Shaohui <liushaohui@xiaomi.com>
Date: Thu, 2 Apr 2015 11:52:54 +0800
Subject: [PATCH 200/226] HBASE-13366 Throw DoNotRetryIOException instead of read only IOException (Shaohui Liu)

Conflicts:
	hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java
(cherry picked from commit fd19f8d44b8d298d750186d05e028c406d71759a)

Reason: Operability
Author: Liu Shaohui
Ref: CDH-26647
---
 .../apache/hadoop/hbase/regionserver/HRegion.java  |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java
index bad0691..4389589 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java
@@ -74,6 +74,7 @@ import org.apache.hadoop.hbase.Cell;
 import org.apache.hadoop.hbase.CellScanner;
 import org.apache.hadoop.hbase.CellUtil;
 import org.apache.hadoop.hbase.CompoundConfiguration;
+import org.apache.hadoop.hbase.DoNotRetryIOException;
 import org.apache.hadoop.hbase.DroppedSnapshotException;
 import org.apache.hadoop.hbase.HBaseConfiguration;
 import org.apache.hadoop.hbase.HColumnDescriptor;
@@ -3530,7 +3531,7 @@ public class HRegion implements HeapSize, PropagatingConfigurationObserver { //
    */
   protected void checkReadOnly() throws IOException {
     if (this.writestate.isReadOnly()) {
-      throw new IOException("region is read only");
+      throw new DoNotRetryIOException("region is read only");
     }
   }
 
-- 
1.7.0.4

